src/control/common/proto/ctl/addons.go

```diff
@@ -49,3 +49,45 @@ func (m *SystemQueryReq) GetSystemRanks() []system.Rank {
 	}
 	return nil
 }
+
+// SetSystemRanks is a convenience method to convert a slice of
+// system ranks to a slice of uint32 ranks for this request.
+func (m *SystemStopReq) SetSystemRanks(sysRanks []system.Rank) error {
+	if m == nil {
+		return errors.New("nil request")
+	}
+	return convert.Types(sysRanks, &m.Ranks)
+}
+
+// GetSystemRanks is a convenience method to convert this request's
+// slice of uint32 ranks to a slice of system ranks.
+func (m *SystemStopReq) GetSystemRanks() []system.Rank {
+	if m != nil {
+		var sysRanks []system.Rank
+		if err := convert.Types(m.GetRanks(), &sysRanks); err == nil {
+			return sysRanks
+		}
+	}
+	return nil
+}
+
+// SetSystemRanks is a convenience method to convert a slice of
+// system ranks to a slice of uint32 ranks for this request.
+func (m *SystemStartReq) SetSystemRanks(sysRanks []system.Rank) error {
+	if m == nil {
+		return errors.New("nil request")
+	}
+	return convert.Types(sysRanks, &m.Ranks)
+}
+
+// GetSystemRanks is a convenience method to convert this request's
+// slice of uint32 ranks to a slice of system ranks.
+func (m *SystemStartReq) GetSystemRanks() []system.Rank {
```

Comment:
my initial inclination was to do as you suggest with a utility function but I didn't want to go against convention that @mjmac set by specifying the methods on the request pointer receiver. will wait for confirmation before making the change.

Reply:
I don't feel strongly either way. It's nice to have them as methods but it does seem to be getting increasingly awkward to have to add them manually. As I noted in another review, I was thinking of these as missing methods that the code generator didn't create. I'll look into whether or not there's any way for us to customize the output, and if not, then perhaps it would be best to consolidate all of these into utility functions.

Reply:
Follow-up: There doesn't seem to be any officially-supported way to customize the code generation, so it's probably best to just extract these into a pair of utility functions as @kjacque suggests.

Reply:
https://github.com/daos-stack/daos/pull/2275/ccommits/03c1b159e9c2bad9dfc280dfd6990049e0e1ee7a & https://github.com/daos-stack/daos/pull/2275/commits/9b2a45af12b4d6d5c242fcddffcae66dcd51518c